packages/web/src/components/admin_page/feature_switch_page/render_switch.jsx

```diff
@@ -0,0 +1,166 @@
+        }));
+    };
+
+    const handleSubmit = e => {
+        e.preventDefault();
+        if (newSwitch.switch_id.trim() !== '') {
+            onUpdate(newSwitch);
+        }
+    };
+
+    return (
+        <Form onSubmit={handleSubmit}>
+            <Form.Group className="mb-3" controlId="formBasicEmail">
+                <Form.Label>Set a feature:</Form.Label>
+                <Form.Control
+                    type="text"
+                    placeholder="Enter new feature switch"
+                    onChange={handleChange}
+                    name="switch_id"
+                    value={newSwitch.switch_id}
+                    required
+                />
+                <Form.Text className="text-muted">Add a new feature</Form.Text>
+            </Form.Group>
+            <Form.Group className="mb-3" controlId="formBasicIsEnabled">
+                <Form.Label>Set it to true or false:</Form.Label>
+                <Form.Select onChange={handleChange} name="is_enabled" value={newSwitch.is_enabled.toString()}>
+                    <option value="true">True</option>
+                    <option value="false">False</option>
+                </Form.Select>
+                <Form.Text className="text-muted"></Form.Text>
+            </Form.Group>
+
+            <Button type="submit" variant="primary">
+                Add
+            </Button>
+        </Form>
+    );
+};
+
+export const popover = (onUpdate, errorMessage) => {
+    return (
+        <Popover id="popover-basic">
+            <Popover.Header as="h3">Want to update or remove?</Popover.Header>
+            <Popover.Body>
+                {errorMessage !== undefined ? <Alert variant="warning">{errorMessage}</Alert> : null}
+                <UpdateSwitch onUpdate={onUpdate} />
+            </Popover.Body>
+        </Popover>
+    );
+};
+
+const updateSwitches = (switches, setSwitches) => data => {
+    setSwitches({ ...switches, [data.switch_id]: data.is_enabled });
+};
+
+const deleteSwitches = (switches, setSwitches) => switchToRemove => {
+    // eslint-disable-next-line no-unused-vars
+    const { [switchToRemove]: _, ...updatedSwitches } = { ...switches }; //creates a new object updatedSwitches that is a copy of switches with the siwtchToRemove, removed
+    setSwitches(updatedSwitches);
+};
+
+const RenderSwitches = () => {
+    const [selected, setSelected] = useState('');
+    const [showAddOverlay, setShowAddOverlay] = useState(false);
+    const [featureSwitches, setFeatureSwitches] = useContext(FsContext);
+    const [errorMessage, setErrorMessage] = useState(undefined);
+    const handleUpdateSwitches = updateSwitches(featureSwitches, setFeatureSwitches);
+    const handleDeleteSwitches = deleteSwitches(featureSwitches, setFeatureSwitches);
+
+    useEffect(() => {
+        axios(featureSwitchesURL, {
+            method: 'get',
+        }).catch(() => {
+            window.alert('API is not responding!');
```

Comment:
Let's handle this with a toast instead of a window message. Hot toast is already being used with the shortener, you may refer to it and its documentation here https://react-hot-toast.com/